src/core/lombok/eclipse/handlers/HandleConstructor.java

```diff
@@ -273,12 +272,20 @@ public void generate(
 		
 		if (noArgs && noArgsConstructorExists(typeNode)) return;
 		
-		ConstructorDeclaration constr = createConstructor(
-			staticConstrRequired ? AccessLevel.PRIVATE : level, typeNode, fieldsToParam, forceDefaults,
-			sourceNode, onConstructor);
-		injectMethod(typeNode, constr);
+		if (skipIfConstructorExists != SkipIfConstructorExists.NO && constructorExists(typeNode) != MemberExistsResult.NOT_EXISTS) {
```

Comment:
Why 
```
if (condition) {
  // generateStatic()
} else {
  // createConstructor
  // generateStatic()
}
```
instead of
```
if (!condition) {
  // createConstructor
}
//generateStatic()

I think the latter is simpler.

Reply:
Greate catch 👍